openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java

```diff
@@ -975,6 +978,47 @@ public final void restoreFromSearch(T entity) {
     }
   }
 
+  public ResultList<T> listFromSearchWithOffset(
+      UriInfo uriInfo,
+      Fields fields,
+      SearchListFilter searchListFilter,
+      int limit,
+      int offset,
+      String q)
+      throws IOException {
+    return listFromSearchWithOffset(
+        uriInfo, fields, searchListFilter, limit, offset, null, null, q);
+  }
+
+  public ResultList<T> listFromSearchWithOffset(
+      UriInfo uriInfo,
+      Fields fields,
+      SearchListFilter searchListFilter,
+      int limit,
+      int offset,
+      String sortField,
+      String sortType,
+      String q)
+      throws IOException {
+    List<T> entityList = new ArrayList<>();
+    Long total = 0L;
+
+    if (limit > 0) {
+      SearchClient.SearchResultListMapper results =
+          searchRepository.listWithOffset(
+              searchListFilter, limit, offset, entityType, sortField, sortType, q);
+      total = results.getTotal();
+      for (Map<String, Object> json : results.getResults()) {
+        T entity = setFieldsInternal(JsonUtils.readOrConvertValue(json, entityClass), fields);
+        setInheritedFields(entity, fields);
+        clearFieldsInternal(entity, fields);
+        entityList.add(withHref(uriInfo, entity));
+      }
+      return new ResultList<>(entityList, offset, limit, total.intValue());
+    }
+    throw new IOException("Limit should be greater than 0");
```

Comment:
this should be IllegalArgumentException, we should reserve IOExceptions for Network/Disk related errors